### PR TITLE
API: recursive tree request

### PR DESCRIPTION
**Pull request by rebecamendez**
As i've suggested in http://feedback.gitlab.com/forums/176466-general/suggestions/6661459-api-allow-recursive-repository-tree: added a parameter in tree request to get a recursive tree and path attribute in response.

### CODE REVIEW

File: app/models/repository.rb [hunk 1]
```diff
@@ -220,7 +220,7 @@ def head_tree
     @head_tree ||= Tree.new(self, head_commit.sha, nil)
   end
 
-  def tree(sha = :head, path = nil)
+  def tree(sha = :head, path = nil, recursive = 0)
```
Comment (jvanbaarsen) [inline app/models/repository.rb]: Why did you choose `0` instead of false?

File: app/models/tree.rb [hunk 2]
```diff
@@ -3,15 +3,26 @@ class Tree
 
   attr_accessor :repository, :sha, :path, :entries
 
-  def initialize(repository, sha, path = '/')
+  def initialize(repository, sha, path = '/', recursive = '0')
```
Comment (jvanbaarsen) [inline app/models/tree.rb]: Is there a reason to be using `'0'` over `false`?

**Comment by rebecamendez** [comment]
I use the Github syntax: https://developer.github.com/v3/git/trees/#get-a-tree-recursively, but if you prefer we can change the flag to a boolean value.

**Comment by jvanbaarsen** [comment]
@rebecamendez Hm ok, but maybe we can translate the URL params to actual booleans where they come in? That way we don't need to pass around `1` and `0`.

**Comment by rebecamendez** [comment]
I made a mistake with the rebase, so I start a new pull request: https://github.com/gitlabhq/gitlabhq/pull/9311

**Pull request closed without merging**